llvm/lib/Transforms/Vectorize/LoadStoreVectorizer.cpp

```diff
@@ -1634,3 +1645,27 @@ std::optional<APInt> Vectorizer::getConstantOffset(Value *PtrA, Value *PtrB,
         .sextOrTrunc(OrigBitWidth);
   return std::nullopt;
 }
+
+void Vectorizer::propagateBestAlignmentsInChain(ArrayRef<ChainElem> C) const {
+  ChainElem BestAlignedElem = C[0];
+  Align BestAlignSoFar = getLoadStoreAlignment(C[0].Inst);
+
+  for (const ChainElem &E : C) {
```

Comment:
We may as well propagate alignment backwards in the chain as well as forwards.

Reply:
Good idea. I'll add a test case and implement this.